docs/plans/token-usage-tracking.md

```
@@ -0,0 +1,331 @@
+## Overview
+
+Implement a dedicated token usage tracking system, separated from audit logs, with a feature flag to enable/disable it.
+
+---
+
+## Research Findings
+
+### Current State Analysis
+
+**1. Core Usage Struct is Minimal** (`internal/core/types.go`)
+```go
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)